rock_paper_scissors/exc.py

```diff
@@ -0,0 +1,49 @@
+class ProgramError(Exception):
+    """Base exception class for errors raised by rock paper scissors program."""
+    pass
+
+
+class UserDefinedError(ProgramError):
+    """Raised for all user input errors."""
+    pass
+
+
+class UserChoiceInputError(UserDefinedError):
+    """Raised for invalid User sting input."""
+    def __init__(self, message):
+        self.message = message
+
+
+class UserRoundsInputError(UserDefinedError):
+    """Raised for invalid rounds input."""
+    pass
+
+
+class RoundsStringError(UserRoundsInputError):
```

Comment:
I see what you mean here. Having multiple sub-classes for one input error may be too much. I could see how this could become much more dense with a larger program attempting, (esp. user input errors) to handle the breadth of errors possible with a single element.